umbral/keys.py

```diff
@@ -87,10 +94,13 @@ def save_key(self, password: bytes=None, _scrypt_cost: int=20):
         encoded_key = base64.urlsafe_b64encode(umbral_priv_key)
         return encoded_key
 
-    def get_pub_key(self, params):
+    def get_pub_key(self, params: UmbralParameters=None):
         """
         Calculates and returns the public key of the private key.
         """
+        if params is None:
+            params = UmbralParameters()
+
```

Comment:
I think we should probably have params as an attribute of Public and Private keys. Consider this:
I create a private key using gen_key with a set of params, and then call get_pub_key with a different set of params. Of course we are not considering different sets of params, but it feels that there is too much freedom in there.

Reply:
Yeah, you're right. I'll switch to this.